OfflineUX.md

```diff
@@ -0,0 +1,138 @@
+
+Used when…
+ - the user needs feedback that data is being sent to the server
+This is a minority use case at the moment, but INCREDIBLY HELPFUL for the user, so proceed with cautious optimism.
+
+How to implement: Use API.write() to implement this pattern. 
+
+# C - Blocking Form UI Pattern
+This pattern blocks the user from interacting with an entire page.
+
+Used when…
+ - a form is used to make a WRITE request to the server and 
+ - server has to do some validation of the parameters that can’t be done in the client or
+ - server response will be unknown so it cannot be done optimistically
+ - If the request is moving money 
+ - If the request is related to security
+
+What: This pattern greys out the submit button on a form and does not allow the form to be submitted. We also show a "You appear offline" message near the bottom of the screen. Importantly, we do let the user fill out the form fields. That data gets saved locally so they don’t have to fill it out again once online.
+
+When: This should be used when we cannot allow the user to submit the form while offline.
+
+How to implement: Use the `<FormAlertWithSubmitButton/>` component.
+
+# D - Full Page Blocking UI Pattern
+This pattern blocks the user from interacting with an entire page.
+
+Used when…
+ - blocking READ is being performed. This occurs when the data that a user sees cannot be stale data and the data can only be displayed after fetching it from the server (eg. Plaid's list of bank accounts)
+ - the app is offline and the data cannot be fetched
+ - an error occurs when fetching the data and the user needs instructions on what to do next
+This should only be used in the most extreme cases when all other options have been completely and utterly exhausted
+
+What: This pattern blocks the user from interacting with an entire page.
+
+How to implement: Wrap the component you're working on in a `<FullPageOfflineBlockingView>` component. 
+
+### UX Pattern Flow Chart
+
+The following flowchart can be used to determine which UX pattern should be used.
+
+![New Expensify Data Flow Chart](/web/OfflineUX_Patterns_Flowchart.png)
+
+### Answering Questions on the Flow Chart
+
+The numbers in this section correlate to the numbers in each decision box above (the diamond shapes).
+
+1. Does the feature interact with the server?
+
+If you’re changing an existing feature, you can open the network tab of dev tools to see if any network requests are being made when you use the feature. If network requests are being made, the answer to this question is YES. Note: sometimes you may see reqeusts that happen to fire at the same time as the feature you're working on, so be sure to double check. 
+If you’re making a new feature, think about whether any data would need to be retrieved or stored from anywhere other than the local device. If data needs to be stored to, or retrieved from the server, then the answer is YES.
+
```

Comment:
Would a negative example where the answer resolves to **NO** be useful?

Reply:
I don't think it's really needed. If you want one here, feel free to write the copy as a suggestion :-)